File: src/main/java/itmo/is/project/mapper/resource/ResourceAmountMapper.java
package itmo.is.project.mapper.resource;

import itmo.is.project.dto.resource.ResourceAmountDto;
import itmo.is.project.mapper.EntityMapper;
import itmo.is.project.model.module.storage.StoredResource;
import itmo.is.project.model.resource.ResourceAmount;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring", uses = {ResourceMapper.class})
public interface ResourceAmountMapper extends EntityMapper<ResourceAmountDto, ResourceAmount> {
    ResourceAmountDto toDto(StoredResource storedResource);
}

File: src/main/kotlin/itmo/isproject/mapper/resource/ResourceAmountMapper.kt
package itmo.isproject.mapper.resource

import itmo.isproject.dto.resource.ResourceAmountDto
import itmo.isproject.mapper.EntityMapper
import itmo.isproject.model.module.storage.StoredResource
import itmo.isproject.model.resource.ResourceAmount
import org.mapstruct.Mapper

@Mapper(uses = [ResourceMapper::class])
interface ResourceAmountMapper : EntityMapper<ResourceAmountDto, ResourceAmount> {

    fun toDto(storedResource: StoredResource): ResourceAmountDto
}
